File: archive/alarm.js
const cp = require('child_process');
const { exit } = require('process');

const playlist_path = `C:\\Users\\BlazingSky\\AppData\\Roaming\\Winamp\\Plugins\\ml\\playlists\\plf8E17.m3u8`
const player_path   = `M:\\Projects\\Development\\Alarm\\tools\\foobar2000\\foobar2000.exe`
const svw_path      = `M:\\Projects\\Development\\Alarm\\tools\\SoundVolumeView.exe`

const initial_volume = 10
const final_volume   = 50

function execute(cmd, silent = true) {
    return new Promise (resolve => {
        cp.exec(cmd, (error, stdout, stderr) => {
            console.log(`${cmd}`)

            if (!silent) {
                if (error) {
                    console.error(error)
                }
                else {
                    console.log(`stdout: ${stdout}`)
                    console.log(`stderr: ${stderr}`)
                }
            }

            resolve()
        })
    })
}

function sleep(ms) {
    return new Promise (resolve => setTimeout(resolve, ms))
}

async function alarm() {
    await execute(`${svw_path} /SetVolume "Headphones" ${initial_volume}`)
    execute(`"${player_path}" "${playlist_path}"`)
    await sleep(5000)

    if (initial_volume < final_volume) {
        for (let x = 0, steps = 40; x <= steps; x++) {
            volume_change = (final_volume - initial_volume) * x / steps
            volume = initial_volume + volume_change
            await execute(`${svw_path} /SetVolume "Headphones" ${volume}`)
            await sleep(1000)
        }
    }

    exit(0)
}

alarm()

File: alarm.py
import subprocess
import time
import sys

# Configure
playlist_path = 'C:\\Users\\BlazingSky\\AppData\\Roaming\\Winamp\\Plugins\\ml\\playlists\\plf8E17.m3u8'
player_path   = 'M:\\Projects\\Development\\Alarm\\tools\\foobar2000\\foobar2000.exe'
svv_path      = 'M:\\Projects\\Development\\Alarm\\tools\\SoundVolumeView.exe'
svv_target    = 'Headphones'

initial_volume = 10
final_volume   = 20

def execute(cmd, wait=True):
  print(cmd)
  if wait:
    subprocess.call(cmd, shell=True)
  else:
    subprocess.Popen(cmd, shell=True)

def main():
  execute(f'{svv_path} /SetVolume "{svv_target}" {initial_volume:.1f}')
  execute(f'"{player_path}" "{playlist_path}"', False)

  if initial_volume < final_volume:
    x     = 0
    steps = final_volume - initial_volume
    while x <= steps:
      volume_change = (final_volume - initial_volume) * x / steps
      volume = initial_volume + volume_change
      execute(f'{svv_path} /SetVolume "{svv_target}" {volume:.1f}')
      time.sleep(4)
      x += 1

start = time.perf_counter()
main()
print(f'Completed in {time.perf_counter() - start} ms')